src/main/php/PHPMD/Rule/CleanCode/UndefinedVariable.php

```diff
@@ -172,17 +160,13 @@ protected function collectListExpressions(AbstractCallableNode $node)
      */
     protected function collectForeachStatements(AbstractCallableNode $node)
     {
-        $foreachStatements = $node->findChildrenOfType('ForeachStatement');
-
-        foreach ($foreachStatements as $foreachStatement) {
-            foreach ($foreachStatement->getChildren() as $children) {
-                if ($children instanceof ASTVariable) {
-                    $this->addVariableDefinition($children);
-                } elseif ($children instanceof ASTUnaryExpression) {
-                    foreach ($children->getChildren() as $refChildren) {
-                        if ($refChildren instanceof ASTVariable) {
-                            $this->addVariableDefinition($refChildren);
-                        }
+        foreach ($node->findChildrenWithParentType('ForeachStatement') as $child) {
+            if ($child instanceof ASTVariable) {
+                $this->addVariableDefinition($child);
+            } elseif ($child instanceof ASTUnaryExpression) {
```

Comment:
Maybe you can lower the indentation by using continue to skip a part of the foreach block.
```suggestion
                continue;
            }
            if (!($child instanceof ASTUnaryExpression)) {
                continue;
            }
```